protocol/app/app.go

```diff
@@ -284,6 +288,14 @@ type App struct {
 	Server              *daemonserver.Server
 }
 
+// assertAppPreconditions assert invariants required for an application to start.
+func assertAppPreconditions() {
+	// Check that the default power reduction is set correctly.
+	if sdk.DefaultPowerReduction.Uint64() != 1_000_000_000_000_000_000 {
```

Comment:
do not cast to a `uint64`. per the [docs](https://pkg.go.dev/math/big#Int.Uint64), 
> If x cannot be represented in a uint64, the result is undefined